frontend/src/scenes/insights/insightDataLogic.ts

```diff
@@ -54,7 +58,17 @@ export const insightDataLogic = kea<insightDataLogicType>([
     path((key) => ['scenes', 'insights', 'insightDataLogic', key]),
 
     connect((props: InsightLogicProps) => ({
-        values: [featureFlagLogic, ['featureFlags'], trendsLogic, ['toggledLifecycles as trendsLifecycles']],
+        values: [
+            insightLogic,
+            ['insight'],
+            featureFlagLogic,
+            ['featureFlags'],
+            trendsLogic,
+            ['toggledLifecycles as trendsLifecycles'],
+            // TODO: need to pass empty query here, as otherwise dataNodeLogic will throw
+            dataNodeLogic({ key: insightVizDataNodeKey(props), query: {} as DataNode }),
+            ['response'],
```

Comment:
@mariusandra (also @pauldambra if you have feedback on this) 👍  / 👎  from you for connecting in this way and then using the selector `const {response} = useValues(insightDataLogic(props))`. I remember you wanted to use a response property on the query to cache the response.

Reply:
I like the idea of hiding the dataNodeLogic behind this one so consumers don't need to care.

Maybe rename it? `['response as insightData']` or something similar